scripts/cfos/make_dataset_data_tables.py

```diff
@@ -87,113 +60,174 @@ def generate_load_table_files(dataset_tables_dict):
+            validated_dataset[table] = dataset[table]
+            print(f"Success: Table {table} has been validated.")
 
-        # TODO: edit the value = dataframe for a table to have the entity:table_name_id format
+    if not errlog.closed:
+        errlog.close()
 
-        # add KVP {table name : table_df} entry to all tables dictionary
-        dataset_tables_dict[table] = table_df
+    if has_err:
+        print(f"Errors found, see {VALIDATION_ERROR_FILE} for details.")
+        exit()
 
-    print(f"Success: Dataset has been partitioned into {len(dataset_table_names)} tables: {dataset_table_names}")
-    return dataset_tables_dict
+    print(f"Success: Dataset has been validated.")
+    return validated_dataset
 
 
-def validate_dataset(dataset):
-    """Validate the full dataset metadata with Schema validator."""
+def col_is_in_defined_fields(col_name, field_dict):
+    list_of_fields = list(field_dict.keys())
+    if col_name in list_of_fields:
+        return True
 
-    # capture errors from the validation against the pre-defined validations per column
-    errors = schema_validator.validate(dataset, columns=schema_validator.get_column_names())
-    # get indices of the failed rows
-    errors_index_rows = [e.row for e in errors]
+    return False
 
-    # if any errors, print error message and exit
-    if errors_index_rows:
-        print(f"Failed: Dataset validation errors found. Please retry after correcting errors listed in validation_errors.csv.")
-        pd.DataFrame({'validation_error':errors}).to_csv('validation_errors.csv')
-        exit()
 
-    # TODO: determine how to handle drops of failed rows/columns to get cleaned data
-    # errors_index_rows = [e.row for e in errors]
-    # data_clean = dataset_metadata_df.drop(index=errors_index_rows, axis=1)
-    # data_clean.to_csv('clean_data.csv')
-    print(f"Success: Dataset has been validated.")
-    return dataset
+def load_excel_input(excel, allowed_dataset_cols, skiprows):
+    """
+    Read excel file input into a dataframe and validate for required columns.
+        Skipping first two rows, as the input excel has two rows of headers before the data starts.
+        Usecols uses a lambda function to filter out columns that aren't defined in the schema.json file's 
+            fields definition section. This is used instead of a list of expected columns, because providing
+            a list enforces that exactly those columns must exist in the data frame, and we don't want to
+            enforce the schema until we're checking it during validation. This should solely be data import.
+    """
 
+    raw_dataset_df = pd.read_excel(excel, sheet_name=None, skiprows=skiprows, index_col=None, usecols=lambda x: x in allowed_dataset_cols)
 
-def load_excel_input(excel, expected_dataset_cols):
-    """Read excel file input into a dataframe and validate for required columns."""
+    print("Success: Excel file has been loaded into a dataframe.")
+    return raw_dataset_df
```

Comment:
I'd rename raw_dataset_df to something like raw_dataset_dict since it's a dictionary, not a dataframe - this was the source of some of my confusion